File: health_advanced_swift/HealthDashboard/Views/HDDashboardCardView.m
//
//  HDDashboardCardView.m
//  HealthDashboard - 卡片容器
//
//  Created by zhang, haizi on 2026/3/19.
//

#import "HDDashboardCardView.h"

@interface HDDashboardCardView ()
@property (nonatomic, strong) UILabel *_titleLabel;
@property (nonatomic, strong) UILabel *_subtitleLabel;
@end

@implementation HDDashboardCardView

- (instancetype)initWithTitle:(NSString *)title iconEmoji:(NSString *)emoji {
    self = [super initWithFrame:CGRectZero];
    if (self) {
        self.backgroundColor    = [UIColor colorWithRed:0.12 green:0.16 blue:0.24 alpha:1.0];
        self.layer.cornerRadius = 16;
        self.layer.shadowColor  = [UIColor blackColor].CGColor;
        self.layer.shadowOpacity = 0.18;
        self.layer.shadowRadius  = 8;
        self.layer.shadowOffset  = CGSizeMake(0, 2);

        // emoji 图标
        UILabel *icon = [[UILabel alloc] initWithFrame:CGRectMake(16, 14, 28, 28)];
        icon.text = emoji;
        icon.font = [UIFont systemFontOfSize:22];
        [self addSubview:icon];

        // 标题
        __titleLabel = [[UILabel alloc] initWithFrame:CGRectMake(50, 14, 200, 22)];
        __titleLabel.text      = title;
        __titleLabel.font      = [UIFont systemFontOfSize:14 weight:UIFontWeightSemibold];
        __titleLabel.textColor = [UIColor whiteColor];
        [self addSubview:__titleLabel];

        // 副标题
        __subtitleLabel = [[UILabel alloc] initWithFrame:CGRectMake(50, 34, 240, 18)];
        __subtitleLabel.font      = [UIFont systemFontOfSize:11];
        __subtitleLabel.textColor = [UIColor colorWithWhite:0.55 alpha:1.0];
        [self addSubview:__subtitleLabel];
    }
    return self;
}

- (UILabel *)titleLabel    { return __titleLabel; }
- (UILabel *)subtitleLabel { return __subtitleLabel; }

- (void)addContentView:(UIView *)view {
    [self addSubview:view];
}

@end

File: health_advanced_swift/HealthDashboard/Views/HDDashboardCardView.swift
//
//  HDDashboardCardView.swift
//  HealthDashboard - 卡片容器
//
//  Created by zhang, haizi on 2026/3/22.
//

import UIKit

/// 通用卡片容器，圆角阴影
class HDDashboardCardView: UIView {

    // MARK: - Properties

    let titleLabel = UILabel()
    let subtitleLabel = UILabel()

    // MARK: - Init

    init(title: String, iconEmoji: String) {
        super.init(frame: .zero)
        backgroundColor = UIColor(red: 0.12, green: 0.16, blue: 0.24, alpha: 1)
        layer.cornerRadius = 16
        layer.shadowColor = UIColor.black.cgColor
        layer.shadowOpacity = 0.18
        layer.shadowRadius = 8
        layer.shadowOffset = CGSize(width: 0, height: 2)

        // emoji 图标
        let icon = UILabel(frame: CGRect(x: 16, y: 14, width: 28, height: 28))
        icon.text = iconEmoji
        icon.font = .systemFont(ofSize: 22)
        addSubview(icon)

        // 标题
        titleLabel.frame = CGRect(x: 50, y: 14, width: 200, height: 22)
        titleLabel.text = title
        titleLabel.font = .systemFont(ofSize: 14, weight: .semibold)
        titleLabel.textColor = .white
        addSubview(titleLabel)

        // 副标题
        subtitleLabel.frame = CGRect(x: 50, y: 34, width: 240, height: 18)
        subtitleLabel.font = .systemFont(ofSize: 11)
        subtitleLabel.textColor = UIColor(white: 0.55, alpha: 1)
        addSubview(subtitleLabel)
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }

    // MARK: - Public

    func addContentView(_ view: UIView) {
        addSubview(view)
    }
}
